### PR TITLE
JIT cleanup

### PR DESCRIPTION
**Pull request by zdevito**
This removes functionality related to @compile that is no longer used:

* Removes interpreter_autograd_function
* handles/handle builder in the interpreter
* creates_handles/tracing_autograd_python_function flags
* var_flags in PythonOp and CppOp

Know to be still remaining but now made dead:
* stage support in the interpreter

### CODE REVIEW

**Comment by apaszke** [comment]
Sorry I’ve been working on the loop unrolling PR. I’ll try to take a look over the weekend

**Review by apaszke — APPROVED** [review]
Looks great! I'm a bit concerned about `THPVariable_Wrap` in `createPythonOp`, so please take a look at that, but ok apart from this.

File: torch/csrc/jit/interpreter.cpp [hunk 1]
```diff
@@ -457,100 +389,49 @@ Operation createPythonOperation(PythonOp* op) {
...
-        }
-        i++;
+    for (auto arg_type : op->cconv) {
+      if (arg_type == 's') {
+        py_inputs[i] = py::reinterpret_borrow<py::object>(
+            op->scalar_args[next_scalar++].get());
+      } else if (arg_type == 't') {
+        auto var = peek(stack, next_tensor, num_inputs);
+        py_inputs[i] =
+            py::reinterpret_steal<py::object>(THPVariable_Wrap(var));
```
[the remark this answers is not in the document]
Comment (zdevito) [reply torch/csrc/jit/interpreter.cpp]: Yes -- my understanding is that THPVariable_Wrap just turns a autograd::Variable into a PyObject* so there is not double-wrapping going on here just conversion into python.

File: torch/csrc/jit/python_tracer.cpp [hunk 2]
```diff
@@ -54,7 +54,7 @@ void initPythonTracerBindings(PyObject* module_) {
     });
 
   m.def("_tracer_enter", [](variable_list trace_inputs, std::size_t num_backwards) {
-    return tracer::enter(std::move(trace_inputs), num_backwards + 1, true);
+    return tracer::enter(std::move(trace_inputs), num_backwards + 1);
```
[the remark this answers is not in the document]
Comment (zdevito) [reply torch/csrc/jit/python_tracer.cpp]: Yep, that should be possible.

File: torch/csrc/jit/interpreter.cpp [hunk 3]
```diff
@@ -444,8 +377,7 @@ bool hasHandleOutput(Node * n) {
 #ifndef NO_PYTHON
 Operation createPythonOperation(PythonOp* op) {
   py::function func = py::reinterpret_borrow<py::function>(py::handle(op->pyobj.get()));
-  bool tracing_autograd_python_function = op->tracing_autograd_python_function;
-  bool has_handle = hasHandleOutput(op);
+  JIT_ASSERT(!hasHandleOutput(op));
```
[the remark this answers is not in the document]
Comment (zdevito) [reply torch/csrc/jit/interpreter.cpp]: Handle type should be dead now, we can remove it a future PR.

File: torch/csrc/jit/interpreter.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (apaszke) [reply torch/csrc/jit/interpreter.cpp]: Uh ofc. I don't know what was I thinking

**Pull request merged**